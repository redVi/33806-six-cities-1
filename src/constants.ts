export const FILTER_OPTIONS = [
  { text: "Popular", value: "popular" },
  { text: "Price: low to high", value: "price", field: "ASC" },
  { text: "Price: high to low", value: "price", field: "DESC" },
  { text: "Top rated first", value: "rating", field: "DESC" }
];

export const STARS = [
  { value: 5, title: "perfect" },
  { value: 4, title: "good" },
  { value: 3, title: "not bad" },
  { value: 2, title: "badly" },
  { value: 1, title: "terribly" },
];
