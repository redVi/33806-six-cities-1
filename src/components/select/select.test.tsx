import React from "react";
import renderer from "react-test-renderer";
import {FILTER_OPTIONS} from "@/constants";
import Select from "@/components/select/select";

describe("Select", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <Select
        opened={true}
        active="Top rated first"
        options={FILTER_OPTIONS}
        handleSelectOption={jest.fn()} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
