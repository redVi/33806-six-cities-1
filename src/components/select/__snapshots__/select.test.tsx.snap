// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex={1}
  >
    Price: low to high
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex={2}
  >
    Price: high to low
  </li>
  <li
    className="places__option places__option--active"
    onClick={[Function]}
    tabIndex={3}
  >
    Top rated first
  </li>
</ul>
`;
